### PR TITLE
Adding motion_interface timeout functionality (#118)

### PR DESCRIPTION
**Pull request by IanTheEngineer**
* Adding timeout functionality to `wait_for_result()` and `send_trajectory()` to block the time before returning.

* Improving documentation and clean-up the code

* Adding documentation to `send_trajectory`

(cherry picked from commit 894c2254543ae68cf9b6beceec3e46be753bf84c)

@asthakukreja and @rethink-forrest, I had to resolve a few conflicts to perform this cherry-pick. Can you please give this a look through to make sure I haven't clobbered anything?

### CODE REVIEW

File: intera_examples/scripts/send_random_trajectory.py [hunk 1]
```diff
@@ -159,17 +162,15 @@ def main():
...
 
         if not args.do_not_send:
-            result = traj.send_trajectory()
-
-            if result is None:
-                rospy.logerr('Trajectory FAILED to send')
-            elif result.result:
-                rospy.loginfo('Motion controller successfully finished the trajectory!')
+            result = traj.send_trajectory(timeout=args.timeout)
+            if result is not None and result.result:
```
Comment (rethink-forrest) [inline intera_examples/scripts/send_random_trajectory.py]: Why did you change the error output here which was not included in #118? It can be helpful to track when it was a network issue versus the trajectory did not complete.

File: intera_interface/src/intera_motion_interface/motion_trajectory.py [hunk 2]
```diff
@@ -95,12 +97,16 @@ def get_state(self):
         """
         return self._client.get_state()
 
-    def wait_for_result(self):
+    def wait_for_result(self, timeout=None):
         """
-        The function will not until the trajectory is finished
-        @return True if the goal finished
+        The function will until the trajectory is finished,
```
Comment (rethink-forrest) [inline intera_interface/src/intera_motion_interface/motion_trajectory.py]: Missing a verb here.

File: intera_interface/src/intera_motion_interface/motion_trajectory.py [hunk 3]
```diff
@@ -86,7 +88,7 @@ def send_trajectory(self, wait_for_result=True):
             return None
         self._check_options()
         self._client.send_trajectory(self.to_msg())
-        return self._client.wait_for_result() if wait_for_result else True
+        return self._client.wait_for_result(timeout) if wait_for_result else None
```
Comment (rethink-forrest) [inline intera_interface/src/intera_motion_interface/motion_trajectory.py]: This function should return True instead of None if not waiting for the result. This was changed in PR #100.

File: intera_examples/scripts/send_random_trajectory.py (hunk 1 above)
Comment (IanTheEngineer) [reply intera_examples/scripts/send_random_trajectory.py]: I didn't intend to change any functionality, so please correct me if I did. I just cleaned up the messy logging statements (which admittedly wasn't necessary).

Comment (IanTheEngineer) [reply intera_examples/scripts/send_random_trajectory.py]: Ah, I see it now. Yep, these lines were confusing when performing the cherry-pick as I had to merge by hand.

Comment (IanTheEngineer) [reply intera_examples/scripts/send_random_trajectory.py]: There must have been changes other than here, which is causing my confusion. If you compare the `development` version and the `intera-testing-5.1` version side by side, you'll see these lines don't mesh. What is the desired functionality / output here?

Comment (rethink-forrest) [reply intera_examples/scripts/send_random_trajectory.py]: Keep the change to add the timeout argument parameter to send_trajectory.

The if-else-statement and logging should stay with what was in the development branch.
